tests/src/test/scala/org/apache/openwhisk/core/entity/test/WhiskEntityTests.scala

```diff
@@ -162,4 +147,78 @@ class WhiskEntityTests extends FlatSpec with ExecHelpers with Matchers {
     val trigger = WhiskTrigger(namespace, name)
     assertType(trigger, "trigger")
   }
+
+  behavior of "WhiskDocumentReader"
+
+  it should "check entityType when deserialize" in {
+    def assertType(d: WhiskEntity, entityType: String) = {
```

Comment:
Make it part of class such that it can be used by other test also and avoid duplication